src/plugins/intel_gpu/src/graph/graph_optimizer/mark_runtime_skippable_nodes.cpp

```diff
@@ -52,6 +55,25 @@ void mark_runtime_skippable_nodes::run(program& p) {
             continue;
         }
 
+        // Check whether consecutive runtime skippable nodes is lower than max count.
```

Comment:
[random spot]
I'm not sure if we really have such long sequences, but if so, then in my opinion the proper solution would be improving add_memory_dependency() logic (e.g., do not iterate over all dependencies recursively, but just reuse an existing dependencies list of the input node and append only new entries or something like that)
@wilson-seok, did you have a chance to try something like that?

Reply:
@sshlyapn Thank you for your opinion. I discussed with Taylor before and she told that add_memory_dependency() is needed to update more efficiently also. But we agreed to make WA first to resolve ticket quickly and consider add_memory_dependency() algorithm later.